### PR TITLE
[Admin][Order Cycle] Enable save btn after removing fee

### LINKED ISSUE #10005: [Admin][Order Cycle] Impossible to remove a fee
## Description
If you add a fee to an order cycle and save, then you can't remove it because the save button is not clickable. 
-- only for shops (it's ok for hubs) --

## Steps to Reproduce
1. Create an enterprise which is "Shop" (you can see the enterprise type in Enterprises tab, column "Package")
2. Create an enterprise fee for this shop (Enterprise > Settings > Enterprise fee)
3. Create or Edit an Order cycle (/admin/order_cycles/15938/edit), add the fee and save
4. Remove the fee and see that you can't save

## Animated Gif/Screenshot
https://user-images.githubusercontent.com/28386648/202262774-5a7404f7-d717-4f03-bc0f-ab34cb312b81.mp4

## Workaround
Create a new OC without the fee.

## Severity
bug-s3: a feature is broken but there is a workaround

### PR DESCRIPTION
**Pull request by binarygit**
#### What? Why?

- Closes #10005 

#### What should we test?

- Login as an enterprise manager who manages a `shop`
- Navigate to  `/admin/order_cycles`
- click edit an order cycle
- Remove an existing fee
- Ascertain that the `save` button becomes enabled

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by binarygit** [comment]
I'm getting this one error from a test which I don't know how to solve. I tried a few things but none worked, I'm not very good with Angular so help is appreciated :smile: 
```
TypeError: Cannot set properties of undefined (setting '$dirty')
            at Object.$scope.removeCoordinatorFee (app/assets/javascripts/admin/order_cycles/controllers/order_cycle_basic_controller.js.coffee:41:7 <- app/assets/javascripts/admin/order_cycles/controllers/order_cycle_basic_controller.js.js:37:43)
            at Object.<anonymous> (spec/javascripts/unit/admin/order_cycles/controllers/order_cycle_basic_controller_spec.js.coffee:61:11 <- spec/javascripts/unit/admin/order_cycles/controllers/order_cycle_basic_controller_spec.js.js:75:11)
```

File: spec/javascripts/unit/admin/order_cycles/controllers/order_cycle_basic_controller_spec.js.coffee [hunk 1]
```diff
@@ -9,6 +9,7 @@ describe 'AdminOrderCycleBasicCtrl', ->
 
   beforeEach ->
     scope =
+      order_cycle_form: jasmine.createSpyObj('order_cycle_form', ['$dirty', '$setPristine'])
```
Comment (filipefurtad0) [inline spec/javascripts/unit/admin/order_cycles/controllers/order_cycle_basic_controller_spec.js.coffee:12]: This is a whole new world for me :eyes:

**Comment by filipefurtad0** [comment]
Hey @binarygit,

I've reproduced the issue: the `Save` and `Save and back to list` buttons was not being activated, when the user has a `sells own produce` profile. If we somehow changed other fields and enabled the buttons, then the change would be effective.

After this PR: the `Save` and `Save and back to list` buttons are enabled after removing an enterprise fee. The issue is solved :+1: 

[Screencast from 14-12-2022 17:11:36.webm](https://user-images.githubusercontent.com/49817236/207661997-51b86fb1-0115-4e0c-bfcb-aa1080f87605.webm)

Good to go!! 

PS: Thanks for that spec @jibees .  I think it's the first time I see a JS spec failing (it probably occurred before, I was just not aware). Interesting! Maybe an interesting approach to replace bulky/slow system specs using?

**Comment by jibees** [comment]
> Thanks for that spec @jibees . I think it's the first time I see a JS spec failing (it probably occurred before, I was just not aware). Interesting! Maybe an interesting approach to replace bulky/slow system specs using?

JS specs are more at a unit test level. I don't see them replacing system ones.

**Pull request merged**